nzedb/NameFixer.php

```diff
@@ -1494,4 +1578,38 @@ public function fileCheck($release, $echo, $type, $namestatus, $show)
 			}
 		}
 	}
+
+	/**
+	 * Look for a name based on mediainfo xml Unique_ID.
+	 *
+	 * @param         $release
+	 * @param         $release
+	 * @param boolean $echo
+	 * @param string  $type
+	 * @param         $namestatus
+	 * @param         $show
+	 */
+	public function uidCheck($release, $echo, $type, $namestatus, $show)
+	{
+		if ($this->done === false && $this->relid !== $release["releases_id"]) {
+			$result = $this->pdo->queryExec(
+				sprintf('
+						SELECT r.id AS releases_id, r.size AS relsize, r.name AS textstring, r.searchname AS searchname, r.predb_id, HEX(ru.uniqueid) AS uid
+						FROM releases r
+						INNER JOIN release_unique ru ON (ru.releases_id = r.id)
+						WHERE (r.isrenamed = %s OR r.categories_id NOT IN(%s)) AND r.predb_id > 0',
+						self::IS_RENAMED_DONE,
+						$this->othercats
+						)
+					);
+			foreach ($result as $rel) {
```

Comment:
After changing the query above, we no longer need to foreach anything.  Either we find a match and run updateRelease with the required parameters or we return false for no match.